src/Common/src/System/Net/HttpKnownHeaderNames.TryGetHeaderName.cs

```diff
@@ -155,6 +155,7 @@ public static unsafe bool TryGetHeaderName(IntPtr buffer, int length, out string
                         case 'K': potentialHeader = KeepAlive; goto TryMatch;  // [K]eep-Alive
                         case 'S': potentialHeader = SetCookie; goto TryMatch;  // [S]et-Cookie
                         case 'U': potentialHeader = UserAgent; goto TryMatch;  // [U]ser-Agent
+                        case 'R': potentialHeader = RequestId; goto TryMatch;  // [R]equest-Id
```

Comment:
@stephentoub 
Apparently, there is a [test](https://ci.dot.net/job/dotnet_corefx/job/master/job/centos7.1_debug_prtest/3750/) that checks all known headers are handled in `TryGetHeaderName`. Do you know if it's intended test behavior?
I'll return back my changes for now, but I'm open to fix the test if needed.